server/game/gameSystems/MoveCardSystem.ts

```diff
@@ -1,19 +1,27 @@
 import type { AbilityContext } from '../core/ability/AbilityContext';
 import type { Card } from '../core/card/Card';
-import { CardType, EffectName, EventName, Location, WildcardCardType } from '../core/Constants';
+import { CardType, EventName, GameStateChangeRequired, Location, WildcardCardType } from '../core/Constants';
+import * as Contract from '../core/utils/Contract';
 import * as EnumHelpers from '../core/utils/EnumHelpers';
+import * as Helpers from '../core/utils/Helpers.js';
 import { type ICardTargetSystemProperties, CardTargetSystem } from '../core/gameSystem/CardTargetSystem';
 
+/**
+ * Properties for moving a card within the game.
+ *
+ * @remarks
+ * Use this interface to specify the properties when moving a card to a new location.
+ * Note that to move cards to the discard pile, any arena, or to the resources, you should use the appropriate systems
+ * such as {@link DiscardSpecificCardSystem}, {@link PlayCardSystem}, or {@link ResourceCardSystem}.
+ *
+ * @property destination - The target location for the card. Excludes discard pile, space arena, ground arena, and resources.
+ * @property shuffle - Indicates whether the card should be shuffled into the destination.
+ * @property bottom - Indicates whether the card should be placed at the bottom of the destination.
+ */
 export interface IMoveCardProperties extends ICardTargetSystemProperties {
-    destination?: Location;
-    switch?: boolean;
-    switchTarget?: Card;
+    destination?: Exclude<Location, Location.SpaceArena | Location.GroundArena | Location.Resource>;
```

Comment:
Discard also should be excluded since that's managed by the various discard systems

Reply:
Unfortunately it can't be done because of this:
https://github.com/SWU-Karabast/forceteki/blob/bc614093334ce3356e4071b94a428deabf701dff/server/game/actions/PlayEventAction.ts#L38-L41